### PR TITLE
doc/ amend & new README.md

### PR DESCRIPTION
**Pull request by Zweihorn**
This PR is **Ready for Review.**

#### Description

The folder `doc` is **amended by creation of file `README.md`** (which was missing and what could have lead  to misunderstandings for the not initiated reader, unfortunately). Furthermore, the new `README.md` will be displayed on the GitHub web GUI file listing as the **main source of information** by default which adds to disambiguation and will greatly **ease the access** to the already existing documentation at least on a web GUI like GitHub.

Intended README.md and result for beholders at:
- https://github.com/Zweihorn/minetest/blob/better-docs/doc/README.md

Hope this helps.

#### Appraisal of this PR for easier understanding:

- Goal of the PR
  1. Provide a **valid entry point** to the Minetest Core Documentation
  2. Provide **disambiguation** to the Minetest Core Documentation
  3. New document itself providing correct information
  4. Ease the life of both the established developers as the new contributors in the making
- How does the PR work?
  - Establishes a new `README.md` which was missing to the `doc` folder of MT core
  - Comprises (in an introductory form) the content of the documentation already present in `doc`
  - Provides further information derived from mature sources of information, including but not limited to the following:
    - minetest.net
    - MT dev wiki
    - Doxygen NL
    - Lua.org
    - linuxhowtos.org
- Does it resolve any reported issue?
  - #13076 
  - https://forum.minetest.net/viewtopic.php?f=7&t=29004
- Does this relate to a goal in [the roadmap](../doc/direction.md)?
  - The very basic needs of devs to not repeat "read the docs" anytime due to mislead contributors asking innocently
  - **Better access to MT in general** and promotion of MT for the greater public
- If not a bug fix, why is this PR needed? What usecases does it solve?
  - based on a technicality this might not truly be a bug fix for the MT code itself
  - however providing false and/or misleading documenation should be called **a bug in documentation**
  - improving is no sin

## To do

Find some MT core reviewer, somehow sometimes ...

EDIT: Goals and To-Do for **reaching beyond** the focus of this specific PR to be found at: #13089

## How to test

Have a look at:
- https://github.com/Zweihorn/minetest/blob/better-docs/doc/README.md

Have a read and a thought at your own deliberation. IMO should easily meet the test requirements as I heavily depend my contribution on work already established within the docs and on the mature Minetest information as provided until today.

Thanks for your time and kind advice in advance.

Keep on the good work and prosper.
🌻

### CODE REVIEW

**Comment by Zweihorn** [comment]
Further remarks:
- The **a.m. task list** hopefully should not hinder the approval of the PR as it **reaches beyond** intentionally.
- Thus the a.m. task list might see copy and paste to **new PR with a long-time schedule** in the future. But first things first.

Any corrections, questions, remarks and advice from your side would be welcome, naturally.

Keep on the good work and prosper.

**Comment by Zweihorn** [comment]
First things first and this PR will get approved hopefully and closed afterwards.

Me going creative here. However, please let me ask if it **would be**:

1. Technicaly reasonable and easily feasible with the automatic GitHub process mgt to make this a standing PR ?
2. Would the usual PR process defined by the MT devs allow such endeavour and would they approve of?
3. When (1) AND (2) = TRUE would it be reasonable to rename this PR after first close to re-open and re-use as a standing PR for each major step in a proposed "better the docs" approach?

Time and advice will show the path forward, I presume.

**Comment by oong819** [comment]
README.android should be renamed to avoid confusion (Android.md?)

**Comment by Zweihorn** [comment]
> README.android should be renamed to avoid confusion (Android.md?)

THX - IMO a very valid proposal. Unfortunately, not the main focus of my PR which seeks to introduce the README.md first.

Certainly, I will put this onto the a.m. list of "Goals and To-Do for reaching beyond the focus of this specific PR" for my next PR if becoming some little maintainer of this part of the greater commitment of the community.

However, I would propose to use the naming `android.md` in small letters, as this would sort this documentation after the capital letters and after the then main `README.md` on most if not all file listings, I presume. Wouldn't you agree?

**Comment by Zweihorn** [comment]
@Emojigit - THX

IMO the contribution by #13080 was **worth to become amended** into this PR and see my additional commit accordingly.

The new text was amended to the README.md as the **new Short List section** above the _existing Content List_. However, the spelling was corrected and the brief file descriptions heavily shortened to meet the 72 characters hard wrap of the text.

IMO this certainly adds to the quality of the document and improves the readability for the more impatient reader of the README.md file itself.

Hope this helps.

**Comment by Zweihorn** [comment]
ready for review

_NOTE: There was no file deleted ref https://github.com/minetest/minetest/pull/13077 as this would be a future task /commits/b67433ce0236262045cd3190842b76097729184a ref #13089_

**Comment by Wuzzy2** [comment]
I skimmed over this and it was easy to understand.

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
+README.md for the Minetest Core Docs
+====================================
```
Comment (numberZero) [inline doc/README.md:2]: Useless header

File: doc/README.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+An introductory README document to the comprised MT core docs folder.
+
+The folder `doc` seeks to comprise the **Minetest Documentation** for
+the MT core development within the Minetest (MT) project and its
+respective Git repository at GitHub.
+<https://github.com/minetest/minetest>.
+
+The documents comprised should be including (but not necessarily be
+limited to) the ones listed in alphabetical order in the following:
```
Comment (numberZero) [inline doc/README.md:14]: Unreadable and hardly useful

File: doc/README.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+# Short List
+
+Technical documentation of the Minetest Engine (Short List)
+
+This folder contains documentation to the Minetest Engine that both the
+MT Developers and the MT Modders may be interested in respectively. The
+documents filed in `doc` are short listed below by brief description.
```
Comment (numberZero) [inline doc/README.md:22]: Useless

File: doc/README.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+* `minetest.6` and `minetestserver.6`: man pages respectively
+
+* `mod_channels.png`: image file on the concept of mod channels
+
+* `protocol.txt`: early draft of the MT communications protocol
+
+* `texture_packs.txt` : structure of a texture pack
+
+* `world_format.txt`: structure of a world format definiton
```
Comment (numberZero) [inline doc/README.md:52]: Empty lines make each item be in a separate list.

File: doc/README.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+* `fst_api.txt`: the Forkspec toolkit
+
+* `lgpl-2.1.txt`: GNU Lesser General Public License, Version 2.1
+
+* `lua_api.txt`: the server-side Lua API.
+
+* `menu_lua_api.txt`: the Lua API calls available to the MT main menu
+
+* `minetest.6` and `minetestserver.6`: man pages respectively
```
Comment (numberZero) [inline doc/README.md:44]: Respectively to what?

File: doc/README.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+* `breakages.md`: breaking changes for the next major version
+
+* `builtin_entities.txt`: list of builtin Lua entities
+
+* `client_lua_api.txt`: an (unstable) doc to the MT client modding API
+
+* `direction.md`: roadmaps of further development
+
+* `fst_api.txt`: the Forkspec toolkit
```
Comment (numberZero) [inline doc/README.md:36]: Fork?

File: doc/README.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+
+* `lgpl-2.1.txt`: GNU Lesser General Public License, Version 2.1
+
+* `lua_api.txt`: the server-side Lua API.
+
+* `menu_lua_api.txt`: the Lua API calls available to the MT main menu
+
+* `minetest.6` and `minetestserver.6`: man pages respectively
+
+* `mod_channels.png`: image file on the concept of mod channels
```
Comment (numberZero) [inline doc/README.md:46]: Yeah, that’s correct, I suppose. But what’s that?

File: doc/README.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+
+[File: builtin_entities.txt](builtin_entities.txt)
+
+## The Minetest Lua Client Modding API Reference
+
+Minetest is a game engine inspired by Minecraft, Infiniminer and others.
+It's licensed under the GNU LGPL 2.1, with artwork generally covered by
+CC BY-SA 3.0. There are two major parts to the system, the first being a
+core engine written in C++, the second being a modding API that exposes
+useful functions to a Lua environment.
```
Comment (numberZero) [inline doc/README.md:81]: What does this have to do with the client modding API reference specifically?

File: doc/README.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+
+The main menu is defined as a formspec by Lua in ../builtin/mainmenu/
+Description of formspec language to show your menu is in lua_api.txt
+
+[File: menu_lua_api.txt](menu_lua_api.txt)
+
+## The Minetest Manual
+
+GROFF template to generate the `minetest.6` man page for games on a
+Unixoid OS.
```
Comment (numberZero) [inline doc/README.md:163]: This is plain wrong. It is the man page itself.

**Review by numberZero — COMMENTED** [review]
Basically, NO. Inaccurate and unreadable.

**Comment by Bituvo** [comment]
Related: #13503

File: doc/README.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+and breaks ties.
+
+## How to Donate?
+
+* More information can be found under **Donate** near the bottom page at
+<https://www.minetest.net/get-involved/>
+
+- - -
+
+No spirit was harmed in the making of this document.
```
Comment (Bituvo) [inline doc/README.md:312]: Please remove for formality.
